librosa/core/pitch.py

```diff
@@ -316,17 +316,19 @@ def piptrack(
     # Do the parabolic interpolation everywhere,
     # then figure out where the peaks are
     # then restrict to the feasible range (fmin:fmax)
-    avg = 0.5 * (S[2:] - S[:-2])
+    avg = 0.5 * (S[..., 2:, :] - S[..., :-2, :])
 
-    shift = 2 * S[1:-1] - S[2:] - S[:-2]
+    shift = 2 * S[..., 1:-1, :] - S[..., 2:, :] - S[..., :-2, :]
 
     # Suppress divide-by-zeros.
     # Points where shift == 0 will never be selected by localmax anyway
     shift = avg / (shift + (np.abs(shift) < util.tiny(shift)))
 
     # Pad back up to the same shape as S
-    avg = np.pad(avg, ([1, 1], [0, 0]), mode="constant")
-    shift = np.pad(shift, ([1, 1], [0, 0]), mode="constant")
+    padding = [(0, 0) for _ in S.shape]
+    padding[-2] = (1, 1)
+    avg = np.pad(avg, padding, mode="constant")
+    shift = np.pad(shift, padding, mode="constant")
```

Comment:
I wonder if these would be better implemented as convolutions?  It would probably change behavior though.

Reply:
Punting this to #1199